<!DOCTYPE html><html><head>
      <title>README</title>
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width, initial-scale=1.0">
      
      <link rel="stylesheet" href="file:///c:\Users\xacfr\.vscode\extensions\shd101wyy.markdown-preview-enhanced-0.8.10\crossnote\dependencies\katex\katex.min.css">
      <link rel="stylesheet" href="file:///c:\Users\xacfr\.vscode\extensions\shd101wyy.markdown-preview-enhanced-0.8.10\crossnote\dependencies\font-awesome\css\all.min.css">
      
      
      
      
      <style>
      code[class*=language-],pre[class*=language-]{color:#333;background:0 0;font-family:Consolas,"Liberation Mono",Menlo,Courier,monospace;text-align:left;white-space:pre;word-spacing:normal;word-break:normal;word-wrap:normal;line-height:1.4;-moz-tab-size:8;-o-tab-size:8;tab-size:8;-webkit-hyphens:none;-moz-hyphens:none;-ms-hyphens:none;hyphens:none}pre[class*=language-]{padding:.8em;overflow:auto;border-radius:3px;background:#f5f5f5}:not(pre)>code[class*=language-]{padding:.1em;border-radius:.3em;white-space:normal;background:#f5f5f5}.token.blockquote,.token.comment{color:#969896}.token.cdata{color:#183691}.token.doctype,.token.macro.property,.token.punctuation,.token.variable{color:#333}.token.builtin,.token.important,.token.keyword,.token.operator,.token.rule{color:#a71d5d}.token.attr-value,.token.regex,.token.string,.token.url{color:#183691}.token.atrule,.token.boolean,.token.code,.token.command,.token.constant,.token.entity,.token.number,.token.property,.token.symbol{color:#0086b3}.token.prolog,.token.selector,.token.tag{color:#63a35c}.token.attr-name,.token.class,.token.class-name,.token.function,.token.id,.token.namespace,.token.pseudo-class,.token.pseudo-element,.token.url-reference .token.variable{color:#795da3}.token.entity{cursor:help}.token.title,.token.title .token.punctuation{font-weight:700;color:#1d3e81}.token.list{color:#ed6a43}.token.inserted{background-color:#eaffea;color:#55a532}.token.deleted{background-color:#ffecec;color:#bd2c00}.token.bold{font-weight:700}.token.italic{font-style:italic}.language-json .token.property{color:#183691}.language-markup .token.tag .token.punctuation{color:#333}.language-css .token.function,code.language-css{color:#0086b3}.language-yaml .token.atrule{color:#63a35c}code.language-yaml{color:#183691}.language-ruby .token.function{color:#333}.language-markdown .token.url{color:#795da3}.language-makefile .token.symbol{color:#795da3}.language-makefile .token.variable{color:#183691}.language-makefile .token.builtin{color:#0086b3}.language-bash .token.keyword{color:#0086b3}pre[data-line]{position:relative;padding:1em 0 1em 3em}pre[data-line] .line-highlight-wrapper{position:absolute;top:0;left:0;background-color:transparent;display:block;width:100%}pre[data-line] .line-highlight{position:absolute;left:0;right:0;padding:inherit 0;margin-top:1em;background:hsla(24,20%,50%,.08);background:linear-gradient(to right,hsla(24,20%,50%,.1) 70%,hsla(24,20%,50%,0));pointer-events:none;line-height:inherit;white-space:pre}pre[data-line] .line-highlight:before,pre[data-line] .line-highlight[data-end]:after{content:attr(data-start);position:absolute;top:.4em;left:.6em;min-width:1em;padding:0 .5em;background-color:hsla(24,20%,50%,.4);color:#f4f1ef;font:bold 65%/1.5 sans-serif;text-align:center;vertical-align:.3em;border-radius:999px;text-shadow:none;box-shadow:0 1px #fff}pre[data-line] .line-highlight[data-end]:after{content:attr(data-end);top:auto;bottom:.4em}html body{font-family:'Helvetica Neue',Helvetica,'Segoe UI',Arial,freesans,sans-serif;font-size:16px;line-height:1.6;color:#333;background-color:#fff;overflow:initial;box-sizing:border-box;word-wrap:break-word}html body>:first-child{margin-top:0}html body h1,html body h2,html body h3,html body h4,html body h5,html body h6{line-height:1.2;margin-top:1em;margin-bottom:16px;color:#000}html body h1{font-size:2.25em;font-weight:300;padding-bottom:.3em}html body h2{font-size:1.75em;font-weight:400;padding-bottom:.3em}html body h3{font-size:1.5em;font-weight:500}html body h4{font-size:1.25em;font-weight:600}html body h5{font-size:1.1em;font-weight:600}html body h6{font-size:1em;font-weight:600}html body h1,html body h2,html body h3,html body h4,html body h5{font-weight:600}html body h5{font-size:1em}html body h6{color:#5c5c5c}html body strong{color:#000}html body del{color:#5c5c5c}html body a:not([href]){color:inherit;text-decoration:none}html body a{color:#08c;text-decoration:none}html body a:hover{color:#00a3f5;text-decoration:none}html body img{max-width:100%}html body>p{margin-top:0;margin-bottom:16px;word-wrap:break-word}html body>ol,html body>ul{margin-bottom:16px}html body ol,html body ul{padding-left:2em}html body ol.no-list,html body ul.no-list{padding:0;list-style-type:none}html body ol ol,html body ol ul,html body ul ol,html body ul ul{margin-top:0;margin-bottom:0}html body li{margin-bottom:0}html body li.task-list-item{list-style:none}html body li>p{margin-top:0;margin-bottom:0}html body .task-list-item-checkbox{margin:0 .2em .25em -1.8em;vertical-align:middle}html body .task-list-item-checkbox:hover{cursor:pointer}html body blockquote{margin:16px 0;font-size:inherit;padding:0 15px;color:#5c5c5c;background-color:#f0f0f0;border-left:4px solid #d6d6d6}html body blockquote>:first-child{margin-top:0}html body blockquote>:last-child{margin-bottom:0}html body hr{height:4px;margin:32px 0;background-color:#d6d6d6;border:0 none}html body table{margin:10px 0 15px 0;border-collapse:collapse;border-spacing:0;display:block;width:100%;overflow:auto;word-break:normal;word-break:keep-all}html body table th{font-weight:700;color:#000}html body table td,html body table th{border:1px solid #d6d6d6;padding:6px 13px}html body dl{padding:0}html body dl dt{padding:0;margin-top:16px;font-size:1em;font-style:italic;font-weight:700}html body dl dd{padding:0 16px;margin-bottom:16px}html body code{font-family:Menlo,Monaco,Consolas,'Courier New',monospace;font-size:.85em;color:#000;background-color:#f0f0f0;border-radius:3px;padding:.2em 0}html body code::after,html body code::before{letter-spacing:-.2em;content:'\00a0'}html body pre>code{padding:0;margin:0;word-break:normal;white-space:pre;background:0 0;border:0}html body .highlight{margin-bottom:16px}html body .highlight pre,html body pre{padding:1em;overflow:auto;line-height:1.45;border:#d6d6d6;border-radius:3px}html body .highlight pre{margin-bottom:0;word-break:normal}html body pre code,html body pre tt{display:inline;max-width:initial;padding:0;margin:0;overflow:initial;line-height:inherit;word-wrap:normal;background-color:transparent;border:0}html body pre code:after,html body pre code:before,html body pre tt:after,html body pre tt:before{content:normal}html body blockquote,html body dl,html body ol,html body p,html body pre,html body ul{margin-top:0;margin-bottom:16px}html body kbd{color:#000;border:1px solid #d6d6d6;border-bottom:2px solid #c7c7c7;padding:2px 4px;background-color:#f0f0f0;border-radius:3px}@media print{html body{background-color:#fff}html body h1,html body h2,html body h3,html body h4,html body h5,html body h6{color:#000;page-break-after:avoid}html body blockquote{color:#5c5c5c}html body pre{page-break-inside:avoid}html body table{display:table}html body img{display:block;max-width:100%;max-height:100%}html body code,html body pre{word-wrap:break-word;white-space:pre}}.markdown-preview{width:100%;height:100%;box-sizing:border-box}.markdown-preview ul{list-style:disc}.markdown-preview ul ul{list-style:circle}.markdown-preview ul ul ul{list-style:square}.markdown-preview ol{list-style:decimal}.markdown-preview ol ol,.markdown-preview ul ol{list-style-type:lower-roman}.markdown-preview ol ol ol,.markdown-preview ol ul ol,.markdown-preview ul ol ol,.markdown-preview ul ul ol{list-style-type:lower-alpha}.markdown-preview .newpage,.markdown-preview .pagebreak{page-break-before:always}.markdown-preview pre.line-numbers{position:relative;padding-left:3.8em;counter-reset:linenumber}.markdown-preview pre.line-numbers>code{position:relative}.markdown-preview pre.line-numbers .line-numbers-rows{position:absolute;pointer-events:none;top:1em;font-size:100%;left:0;width:3em;letter-spacing:-1px;border-right:1px solid #999;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.markdown-preview pre.line-numbers .line-numbers-rows>span{pointer-events:none;display:block;counter-increment:linenumber}.markdown-preview pre.line-numbers .line-numbers-rows>span:before{content:counter(linenumber);color:#999;display:block;padding-right:.8em;text-align:right}.markdown-preview .mathjax-exps .MathJax_Display{text-align:center!important}.markdown-preview:not([data-for=preview]) .code-chunk .code-chunk-btn-group{display:none}.markdown-preview:not([data-for=preview]) .code-chunk .status{display:none}.markdown-preview:not([data-for=preview]) .code-chunk .output-div{margin-bottom:16px}.markdown-preview .md-toc{padding:0}.markdown-preview .md-toc .md-toc-link-wrapper .md-toc-link{display:inline;padding:.25rem 0}.markdown-preview .md-toc .md-toc-link-wrapper .md-toc-link div,.markdown-preview .md-toc .md-toc-link-wrapper .md-toc-link p{display:inline}.markdown-preview .md-toc .md-toc-link-wrapper.highlighted .md-toc-link{font-weight:800}.scrollbar-style::-webkit-scrollbar{width:8px}.scrollbar-style::-webkit-scrollbar-track{border-radius:10px;background-color:transparent}.scrollbar-style::-webkit-scrollbar-thumb{border-radius:5px;background-color:rgba(150,150,150,.66);border:4px solid rgba(150,150,150,.66);background-clip:content-box}html body[for=html-export]:not([data-presentation-mode]){position:relative;width:100%;height:100%;top:0;left:0;margin:0;padding:0;overflow:auto}html body[for=html-export]:not([data-presentation-mode]) .markdown-preview{position:relative;top:0;min-height:100vh}@media screen and (min-width:914px){html body[for=html-export]:not([data-presentation-mode]) .markdown-preview{padding:2em calc(50% - 457px + 2em)}}@media screen and (max-width:914px){html body[for=html-export]:not([data-presentation-mode]) .markdown-preview{padding:2em}}@media screen and (max-width:450px){html body[for=html-export]:not([data-presentation-mode]) .markdown-preview{font-size:14px!important;padding:1em}}@media print{html body[for=html-export]:not([data-presentation-mode]) #sidebar-toc-btn{display:none}}html body[for=html-export]:not([data-presentation-mode]) #sidebar-toc-btn{position:fixed;bottom:8px;left:8px;font-size:28px;cursor:pointer;color:inherit;z-index:99;width:32px;text-align:center;opacity:.4}html body[for=html-export]:not([data-presentation-mode])[html-show-sidebar-toc] #sidebar-toc-btn{opacity:1}html body[for=html-export]:not([data-presentation-mode])[html-show-sidebar-toc] .md-sidebar-toc{position:fixed;top:0;left:0;width:300px;height:100%;padding:32px 0 48px 0;font-size:14px;box-shadow:0 0 4px rgba(150,150,150,.33);box-sizing:border-box;overflow:auto;background-color:inherit}html body[for=html-export]:not([data-presentation-mode])[html-show-sidebar-toc] .md-sidebar-toc::-webkit-scrollbar{width:8px}html body[for=html-export]:not([data-presentation-mode])[html-show-sidebar-toc] .md-sidebar-toc::-webkit-scrollbar-track{border-radius:10px;background-color:transparent}html body[for=html-export]:not([data-presentation-mode])[html-show-sidebar-toc] .md-sidebar-toc::-webkit-scrollbar-thumb{border-radius:5px;background-color:rgba(150,150,150,.66);border:4px solid rgba(150,150,150,.66);background-clip:content-box}html body[for=html-export]:not([data-presentation-mode])[html-show-sidebar-toc] .md-sidebar-toc a{text-decoration:none}html body[for=html-export]:not([data-presentation-mode])[html-show-sidebar-toc] .md-sidebar-toc .md-toc{padding:0 16px}html body[for=html-export]:not([data-presentation-mode])[html-show-sidebar-toc] .md-sidebar-toc .md-toc .md-toc-link-wrapper .md-toc-link{display:inline;padding:.25rem 0}html body[for=html-export]:not([data-presentation-mode])[html-show-sidebar-toc] .md-sidebar-toc .md-toc .md-toc-link-wrapper .md-toc-link div,html body[for=html-export]:not([data-presentation-mode])[html-show-sidebar-toc] .md-sidebar-toc .md-toc .md-toc-link-wrapper .md-toc-link p{display:inline}html body[for=html-export]:not([data-presentation-mode])[html-show-sidebar-toc] .md-sidebar-toc .md-toc .md-toc-link-wrapper.highlighted .md-toc-link{font-weight:800}html body[for=html-export]:not([data-presentation-mode])[html-show-sidebar-toc] .markdown-preview{left:300px;width:calc(100% - 300px);padding:2em calc(50% - 457px - 300px / 2);margin:0;box-sizing:border-box}@media screen and (max-width:1274px){html body[for=html-export]:not([data-presentation-mode])[html-show-sidebar-toc] .markdown-preview{padding:2em}}@media screen and (max-width:450px){html body[for=html-export]:not([data-presentation-mode])[html-show-sidebar-toc] .markdown-preview{width:100%}}html body[for=html-export]:not([data-presentation-mode]):not([html-show-sidebar-toc]) .markdown-preview{left:50%;transform:translateX(-50%)}html body[for=html-export]:not([data-presentation-mode]):not([html-show-sidebar-toc]) .md-sidebar-toc{display:none}
/* Please visit the URL below for more information: */
/*   https://shd101wyy.github.io/markdown-preview-enhanced/#/customize-css */

      </style>
      <!-- The content below will be included at the end of the <head> element. --><script type="text/javascript">
  document.addEventListener("DOMContentLoaded", function () {
    // your code here
  });
</script></head><body for="html-export">
    
    
      <div class="crossnote markdown-preview  ">
      
<h1 id="genome-annotation">Genome Annotation </h1>
<p><a href="https://hackmd.io/jqXB_tEgQtWxRVZc1ChNeQ"><img src="https://hackmd.io/jqXB_tEgQtWxRVZc1ChNeQ/badge" alt="hackmd-github-sync-badge"></a></p>
<h6 id="tags-genome-annotation-singularity-braker3">tags: <code>genome annotation</code> <code>singularity</code> <code>BRAKER3</code> </h6>
<h6 id="last-updated-nov-30-2023">Last updated: Nov 30, 2023 </h6>
<hr>
<p>This tutorial was written for students of the Genomes and Genome Evolution course at Texas Tech and tested at the <a href="https://https://www.depts.ttu.edu/hpcc/">High Performance Computing Cluster</a> (HPCC) of this University.</p>
<h2 id="i-classfa-fa-download-fa-lgi-pre-requisites"><i class="fa fa-download fa-lg"></i> Pre-requisites </h2>
<p>You will only need to have Singularity installed. Singularity is a tool that allows you to create and run applications in containers. What are containers you may ask? Of course the name gives away what they do, they contain things. In this case, all the software and dependencies you need to run a specific application. This is AWESOME because you only need to have a <strong>S</strong>ingularity <strong>I</strong>mage <strong>F</strong>ormat (SIF) and anything else. This is great because otherwise it may take you months to install everything you need, I know this from experience. Take a look at <a href="https://github.com/Gaius-Augustus/BRAKER#braker">this</a> and you'll see everything you would need to manually install to use BRAKER3 to annotate a genome.</p>
<p>Luckily for us, Singularity is already installed at the HPCC. If you are using your own computer, you can follow the instructions <a href="https://docs.sylabs.io/guides/3.0/user-guide/installation.html">here</a> to install it.</p>
<p>Lastly, the files you will use in this tutorial are available at my <a href="https://github.com/Xacfran/Genome-Annotation-with-Braker3">GitHub repository</a> if you are not a Tech student. If you are working on the HPCC, you can copy them to your own directory with:</p>
<pre data-role="codeBlock" data-info="bash" class="language-bash bash"><code><span class="token function">cp</span> /lustre/work/frcastel/software/braker/scaffold_myo_myo.fasta <span class="token builtin class-name">.</span>
<span class="token function">cp</span> /lustre/work/frcastel/software/braker/filtered_mammalia.fasta <span class="token builtin class-name">.</span>
</code></pre><p>The first file is the largest scaffold of the genome assembly of the bat <em>Myotis myotis</em> you worked with some weeks ago, and the second one is a filtered version (3,835,124 proteins) of the <a href="https://bioinf.uni-greifswald.de/bioinf/partitioned_odb11/">Vertebrata dataset</a> from <a href="https://www.orthodb.org/">OrthoDB</a>. We'll use this dataset to train the gene models, but first lets get the file that will allow us to use BRAKER3.</p>
<h2 id="i-classfa-fa-gears-fa-lgi-installation"><i class="fa fa-gears fa-lg"></i> Installation </h2>
<p>To get the BRAKER3 SIF file you simply need to run:</p>
<pre data-role="codeBlock" data-info="bash" class="language-bash bash"><code>singularity build braker3.sif docker://teambraker/braker3:latest
</code></pre><h2 id="i-classfa-fa-code-fa-lgi-usage"><i class="fa fa-code fa-lg"></i> Usage </h2>
<p>BRAKER3 runs in different modes, it can make use of the genome alone, genome + proteins, genome + RNA-seq, or all of them. You can find the details of how each mode works <a href="https://github.com/Gaius-Augustus/BRAKER#overview-of-modes-for-running-braker">here</a>. In this tutorial we will use proteins, but first lets use some test scripts to make sure it works.</p>
<h3 id="test-data">Test data </h3>
<pre data-role="codeBlock" data-info="bash" class="language-bash bash"><code>singularity <span class="token builtin class-name">exec</span> <span class="token parameter variable">-B</span> <span class="token environment constant">$PWD</span><span class="token builtin class-name">:</span><span class="token environment constant">$PWD</span> braker3.sif <span class="token function">cp</span> /opt/BRAKER/example/singularity-tests/test1.sh <span class="token builtin class-name">.</span>
singularity <span class="token builtin class-name">exec</span> <span class="token parameter variable">-B</span> <span class="token environment constant">$PWD</span><span class="token builtin class-name">:</span><span class="token environment constant">$PWD</span> braker3.sif <span class="token function">cp</span> /opt/BRAKER/example/singularity-tests/test2.sh <span class="token builtin class-name">.</span>
singularity <span class="token builtin class-name">exec</span> <span class="token parameter variable">-B</span> <span class="token environment constant">$PWD</span><span class="token builtin class-name">:</span><span class="token environment constant">$PWD</span> braker3.sif <span class="token function">cp</span> /opt/BRAKER/example/singularity-tests/test3.sh <span class="token builtin class-name">.</span>
</code></pre><p>This will download 3 test scripts that will run BRAKER3 in different modes. For this tutorial we will only use the first one, but you can try the others on your own. You can run the script with <code>bash test1.sh</code>.</p>
<p>After some minutes you will see a directory called <code>test1</code> with the results of the annotation. You can take a look at the files with <code>ls test1</code>. If you see the files <code>braker.gtf</code>, <code>braker.aa</code> and <code>braker.codingseq</code>, and if they are not empty, then everything worked fine. Now moving on with our own data.</p>
<h3 id="annotating-a-bat-genome-assembly">Annotating a bat genome assembly </h3>
<p>As discussed in class, annotating a genome is a complex process that invokes several pieces of software, each one of them with specifical tasks. BRAKER3 is a pipeline that automates this process so the code looks as simple as any other you have used during the semester as you can see below.</p>
<pre data-role="codeBlock" data-info="bash" class="language-bash bash"><code><span class="token shebang important">#!/bin/bash</span>
<span class="token comment">#SBATCH --job-name=myo_braker</span>
<span class="token comment">#SBATCH --output=%x.%j.out</span>
<span class="token comment">#SBATCH --error=%x.%j.err</span>
<span class="token comment">#SBATCH --partition=nocona</span>
<span class="token comment">#SBATCH --nodes=1</span>
<span class="token comment">#SBATCH --ntasks=36</span>
<span class="token comment">#SBATCH --mem=0</span>

<span class="token comment"># Get the SIF file into the path</span>
<span class="token builtin class-name">export</span> <span class="token assign-left variable">BRAKER_SIF</span><span class="token operator">=</span>/path/to/your/braker3.sif

<span class="token comment"># Create the working directory or delete it if it exists</span>
<span class="token comment"># You can change the name myo_myo to anyting you think is appropriate</span>
<span class="token assign-left variable">WORKDIR</span><span class="token operator">=</span>/path/to/your/working/directory
<span class="token assign-left variable">WD</span><span class="token operator">=</span>myo_myo


<span class="token keyword keyword-if">if</span> <span class="token punctuation">[</span> <span class="token parameter variable">-d</span> <span class="token variable">$WD</span> <span class="token punctuation">]</span><span class="token punctuation">;</span> <span class="token keyword keyword-then">then</span>
    <span class="token function">rm</span> <span class="token parameter variable">-r</span> <span class="token variable">$WD</span><span class="token punctuation">;</span> <span class="token keyword keyword-else">else</span>
    <span class="token function">mkdir</span> <span class="token variable">$WD</span>
<span class="token keyword keyword-fi">fi</span>

<span class="token comment"># Run singularity</span>
singularity <span class="token builtin class-name">exec</span> <span class="token parameter variable">-B</span> <span class="token variable">${WORKDIR}</span><span class="token builtin class-name">:</span><span class="token variable">${WORKDIR}</span> <span class="token variable">${BRAKER_SIF}</span> braker.pl <span class="token punctuation">\</span>
<span class="token parameter variable">--genome</span><span class="token operator">=</span><span class="token variable">$WORKDIR</span>/scaffold_myo_myo.fasta <span class="token punctuation">\</span>
<span class="token parameter variable">--prot_seq</span><span class="token operator">=</span><span class="token variable">$WORKDIR</span>/filtered_mammalia.fasta <span class="token punctuation">\</span>
<span class="token parameter variable">--softmasking</span> <span class="token punctuation">\</span>
<span class="token parameter variable">--workingdir</span><span class="token operator">=</span><span class="token variable">${WORKDIR}</span> <span class="token punctuation">\</span>
<span class="token parameter variable">--threads</span> <span class="token variable">$SLURM_NTASKS</span> <span class="token punctuation">\</span>
<span class="token parameter variable">--gff3</span>
</code></pre><p>Simple, right? Let's break it down. It is extremely important that you understand what each of the arguments in <code>braker.pl</code> do, and what others are available to you. You can find a large list of those arguments by running <code>singularity exec -B ${PWD}:${PWD} ${BRAKER_SIF} braker.pl --help</code>.</p>
<p>The <code>INPUT FILE OPTIONS</code> section contains all arguments related to the data you will give BRAKER3 to do the genome annotation. The first option is of course the genome assembly, which you provide with the <code>--genome</code> argument, <code>--bam</code> can be used if you had bam files of RNA-Seq alignments, and <code>--prot_seq</code> if you have some proteins. The proteins you choose to use here will make a huge impact on the genome annotation, and eventhough BRAKER3 does a fairly good job when using protein evidence of distantly related species, it is always better to use proteins from closely related ones. The <code>--hints</code> argument will be used only if you have any sort information regarding the intron structure of the species (or closely related), and/or if you trained a gene model with any other software before hand. Last arguments available for data input are for raw RNA-Seq which are automatically downloaded from the <a href="https://www.ncbi.nlm.nih.gov/sra">SRA</a>. I encountered some problems with those flags when I tried to use them, so we will not cover it in class. However, I was able to annotate that scaffold using RNA-Seq data of the liver, brain and kidney of <em>Myotis myotis</em>. You will have a look at those results to do some comparisons later.</p>
<p>The other sections of the help page contain arguments related to how you want the algorithm to behave and I reccommend you have a close look to all of them if you are performing a genome annotation of your own. Especially in non-model organisms. The arguments we are using from all of those options are <code>--softmasking</code> which lets the gene finders know that we have marked repetitive regions in the DNA with low-case letters (a,t,c,g), <code>--threads</code> to speed up the process using all cores available. I use <code>$SLURM_NTASKS</code> in there because it takes the number that we provide in <code>#SBATCH --ntasks=36</code> and uses all those processors by default. Finally, <code>--gff3</code> is a personal preference to get a <code>gff3</code> besides the <code>gtf</code> file that BRAKER3 gives as default. For me, it is much easier to transform <code>gff3</code> files into any other format and to extract information that we are interested in from there.</p>
<p>This example took around 7 hours to complete, so make sure you start working on it soon! Besides the <code>braker.gtf</code>, <code>braker.aa</code> and <code>braker.codingseq</code> files you will find in your <code>myo_myo</code> directory, you can also see in the last lines of your err file something like:</p>
<pre data-role="codeBlock" data-info="" class="language-text text"><code>[Wed Nov 22 05:57:01 2023] Finished spliced alignment
[Wed Nov 22 05:57:02 2023] Flagging top chains
[Wed Nov 22 05:57:03 2023] Processing the output
[Wed Nov 22 05:57:07 2023] Output processed
[Wed Nov 22 05:57:07 2023] ProtHint finished.
</code></pre><h2 id="for-you-to-do">For you to do </h2>
<p>Once the genome annotation has completed, use the skills you have acquired in this course to answer the questions below. All these questions can be answered with the commands you have learned in this course. If you need a refresher, go to <a href="https://github.com/davidaray/Genomes-and-Genome-Evolution/wiki/01.-Logging-In-to-HPCC-and-an-Intro-to-Bash-and-Linux-Navigation">Tutorial 1, Logging In to HPCC and an Intro to Bash and Linux Navigation</a>.</p>
<ol>
<li>
<p>How many genes were annotated in this scaffold of the <em>Myotis myotis</em> genome?</p>
</li>
<li>
<p>Based on your answer to the previous question, what is the gene density of this scaffold? That is, gene number/genome size. In this case, this scaffold is 223,369,599 bp long.</p>
</li>
<li>
<p>Are all of them protein coding genes? You can compare the braker.aa and braker.gff3 files to answer this question.</p>
</li>
<li>
<p>Because of time limits we couldn't work on functional annotation of the genes, but you can still get an idea of what functions the annotated genes may have. Use the <code>braker.aa</code> file and randomly pick a protein sequence (Please avoid using the first gene "g1.t1"). You can use the <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE=Proteins">BLASTp</a> tool to search for homologous proteins in the <a href="https://www.ncbi.nlm.nih.gov/refseq/about/nonredundantproteins/">NCBI non-redundant protein database</a>. You can use the default parameters for the BLAST search, so just copy the protein sequence and let the web-tool do its work.<br>
From my experience, though, it is way better to use the <a href="https://www.ebi.ac.uk/interpro/search/sequence/">InterProScan</a> tool to search for protein domains. So, again use the same protein you blasted and search for protein domains in the InterPro web page.<br>
Now, if you were to report this gene in a paper, what would you say about it? How many exons and introns does it have? What's the length of that gene? What's its most likely function?<br>
Please attach a screenshot of the NCBI, Interpro, section of the braker.gff3 file, and anything else you used to come up with your answer.</p>
</li>
<li>
<p>Copy the files I obtained when annotating the genome using RNA-Seq data. You can do this with: <code>cp /lustre/work/frcastel/software/braker/myo_myo_rna/braker_rna.* .</code>. Answer the same questions as in 1 and 2. Are the annotations done with genome + proteins, and genome + proteins + RNA-Seq data similar? Why do you think that is?</p>
</li>
</ol>
<h2 id="sources-and-further-reading">Sources and further reading </h2>

      </div>
      
      
    
    
    
    
    
    
  
    </body></html>